pkg/monitortests/kubeapiserver/installerpod/monitortest.go

```diff
@@ -0,0 +1,294 @@
+package installerpod
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	MonitorName = "installer-pod-monitor"
+)
+
+func NewInstallerPodMonitorTest() monitortestframework.MonitorTest {
+	return &monitorTest{
+		monitor: &installerPodMonitor{
+			pods: map[string]*podInfo{},
+		},
+		filter: func(interval monitorapi.Interval) bool {
+			if ns, ok := interval.Locator.Keys[monitorapi.LocatorNamespaceKey]; !ok || ns != "openshift-etcd" {
```

Comment:
imo this should be moved to openshift-etcd and renamed to include "etcd" prefix as we may want to later expand this to other installers like kube-apiserver